binding_generator.py

```diff
@@ -1545,6 +1553,55 @@ def generate_engine_classes_bindings(api, output_dir, use_template_get_node):
             header_file.write("\n".join(result))
 
 
+def generate_compat_includes(godot_repo: Path, output_dir: Path, target_dir: Path):
+    # Uncomment this to change include files
+    # target_dir = output_dir
+    file_types_mapping_godot_cpp_gen = map_header_files(target_dir)
+    file_types_mapping_godot = map_header_files(godot_repo)
+    # Match the headers
+    file_types_mapping = match_headers(file_types_mapping_godot_cpp_gen, file_types_mapping_godot)
+    for file_godot_cpp_name, file_godot_names in file_types_mapping.items():
+        result = []
+        with (Path(target_dir) / Path(file_godot_cpp_name)).open("r", encoding="utf-8") as header_file:
+            content = header_file.readlines()
+        # Find the last line (#endif guard)
+        last_endif_idx = len(content) - 1
+        # Look for the marker for the header guard (usually #define)
+        marker_pos = next((i for i, line in enumerate(content) if line.startswith("#define")), -1)
+
+        if marker_pos != -1:
+            # Add content before the last #endif
+            before_marker = content[: marker_pos + 1]  # Include up to the marker line
+            after_marker = content[marker_pos + 1 : last_endif_idx]  # Content excluding the final #endif
+
+            result.extend(before_marker)  # Append original content up to marker
+            result.append("\n")  # Blank line for separation
+
+            # Insert the #ifdef block
+            result.append("#ifdef GODOT_MODULE_COMPAT\n")
```

Comment:
Is there an advantage to using `GODOT_MODULE_COMPAT` rather than just `GODOT_MODULE`? When building Godot, it will define `GODOT_MODULE` in its build system, so we don't need to add a new define.

Reply:
Didn't know about the define, will update.